elftools/dwarf/locationlists.py

```diff
@@ -56,21 +154,69 @@ def _parse_location_list_from_stream(self):
                 break
             elif begin_offset == self._max_addr:
                 # Base address selection entry
-                lst.append(BaseAddressEntry(entry_offset=entry_offset, base_address=end_offset))
+                entry_length = self.stream.tell() - entry_offset
+                lst.append(BaseAddressEntry(entry_offset=entry_offset, entry_length = entry_length, base_address=end_offset))
             else:
                 # Location list entry
                 expr_len = struct_parse(
                     self.structs.Dwarf_uint16(''), self.stream)
                 loc_expr = [struct_parse(self.structs.Dwarf_uint8(''),
                                          self.stream)
                                 for i in range(expr_len)]
+                entry_length = self.stream.tell() - entry_offset
                 lst.append(LocationEntry(
                     entry_offset=entry_offset,
+                    entry_length = entry_length,
                     begin_offset=begin_offset,
                     end_offset=end_offset,
-                    loc_expr=loc_expr))
+                    loc_expr=loc_expr,
+                    is_absolute = False))
```

Comment:
here too in this `LocationEntry`,

Also in general not sure the `entry_length` here adds much beyond just inlining the expression into the call (in the previous chunk too)

Reply:
Readelf..loc dumps the offset of the list terminator entry, which our list parser doesn't return. Previously, we'd calculate that offset from the offset and the datatype of the last entry, but with various entry encodings in the picture, that logic becomes too bulky to maintain - I'd have to re-encode the entry to its original representation to recover the length. It's easier to capture the entry length explicitly.

Reply:
Note the bit at readelf.py:1488 - that's where `entry_length` is needed.